src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs

```diff
@@ -413,17 +417,24 @@ public async Task SaveChangesAsync(IList<UnspentOutputs> unspentOutputs, IEnumer
 
                         // We take the original items that are in cache and put them in rewind data.
                         clone.Outputs = cacheItem.UnspentOutputs.Outputs.ToArray();
+                        this.logger.LogTrace("Adding {0}:'{1}' to {2}", nameof(clone), clone, nameof(rewindData.OutputsToRestore));
                         rewindData.OutputsToRestore.Add(clone);
 
                         // Now modify the cached items with the mutated data.
+                        this.logger.LogTrace("{0}:'{1}' not null", nameof(cacheItem.UnspentOutputs), cacheItem.UnspentOutputs);
```

Comment:
at this point its useless to trace `cacheItem.UnspentOutputs` we want to do it after we call `cacheItem.UnspentOutputs.Spend(unspent);` so we capture the change

Reply:
ah I see instead you trace out the `unspent` so no need from one of this lines